### PR TITLE
fix resolve KID in JWS

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: packers/jws.go [hunk 1]
```diff
@@ -74,10 +74,9 @@ type JWSPacker struct {
 
 // SigningParams packer parameters for jws generation
 type SigningParams struct {
-	SenderDIDstr string
```
Comment (ilya-korotya) [inline packers/jws.go:77]: In some cases we may not have the complete did document

File: packers/jws.go [hunk 2]
```diff
@@ -238,77 +227,97 @@ func (p *JWSPacker) Unpack(envelope []byte) (*iden3comm.BasicMessage, error) {
...
 			}
 		} else {
-			blockchainAccountIDParts := strings.Split(vm.BlockchainAccountID, ":")
-			address := blockchainAccountIDParts[len(blockchainAccountIDParts)-1]
+			base64Token, err := jws.Compact(token)
+			if err != nil {
+				continue
+			}
+			base64TokenParts := strings.Split(string(base64Token), ".")
+			signedData := base64TokenParts[0] + "." + base64TokenParts[1]
```
Comment (ilya-korotya) [inline packers/jws.go:279]: Panic is possible here. Please, check len to be sure that `base64TokenParts` variable has needed elements.

File: packers/jws.go [hunk 3]
```diff
@@ -238,77 +227,97 @@ func (p *JWSPacker) Unpack(envelope []byte) (*iden3comm.BasicMessage, error) {
...
+			hash := sha256.Sum256([]byte(signedData))
+			sig := token.Signatures()[0].Signature()
+			recoveredKey, err := ecc.RecoverEthereum(hash[:], sig)
+			if err != nil {
+				continue
+			}
 
-			if !strings.EqualFold(address, ethAddress) {
-				return nil, errors.New("invalid signature from blockchain account id")
+			ethAddress := "0x" + hex.EncodeToString(keccak256.Hash(recoveredKey[1:])[12:])
```
[the remark this answers is not in the document]
Comment (ilya-korotya) [reply packers/jws.go:287]: Here no panic if blockchainAccountIDParts is empty
https://play.golang.com/p/_pKGI-iJNNi

File: packers/jws.go [hunk 4]
```diff
@@ -238,77 +227,97 @@ func (p *JWSPacker) Unpack(envelope []byte) (*iden3comm.BasicMessage, error) {
...
+				continue
+			}
+			_, err = jws.Verify(envelope, wk)
+			if err != nil {
+				continue
 			}
 		} else {
-			blockchainAccountIDParts := strings.Split(vm.BlockchainAccountID, ":")
-			address := blockchainAccountIDParts[len(blockchainAccountIDParts)-1]
+			base64Token, err := jws.Compact(token)
```
Comment (ilya-korotya) [inline packers/jws.go:274]: Can we move this code from `else` to separate function?

File: packers/jws.go (hunk 1 above)
Comment (vmidyllic) [reply packers/jws.go:77]: ???

Comment (ilya-korotya) [reply packers/jws.go:77]: @vmidyllic I remember we discussed that users can have only DID not the whole DID document. But now we removed this possibility.

**Review by ilya-korotya — APPROVED** [review]

**Pull request merged**